src/main/java/hypersquare/hypersquare/dev/code/player/event/PlayerDamagePlayerEvent.java

```diff
@@ -0,0 +1,51 @@
+package hypersquare.hypersquare.dev.code.player.event;
+
+import hypersquare.hypersquare.dev.action.CancellableEvent;
+import hypersquare.hypersquare.dev.target.Target;
+import hypersquare.hypersquare.item.action.player.PlayerEventItems;
+import hypersquare.hypersquare.item.event.EventItem;
+import hypersquare.hypersquare.util.color.Colors;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+
+public class PlayerDamagePlayerEvent implements CancellableEvent {
+    @Override
+    public ItemStack item() {
+        return new EventItem()
+            .setMaterial(Material.IRON_SWORD)
+            .setName(Component.text("Player Damage Player Event").color(Colors.RED_LIGHT))
+            .setDescription(
+                Component.text("Executes code when a player"),
+                Component.text("damages another player."))
+            .setCancellable(true)
+            .build()
+            ;
+    }
+
+    @Override
+    public String getId() {
+        return "player_damage_player";
+    }
+
+    @Override
+    public String getCodeblockId() {
+        return "player_event";
+    }
+
+    @Override
+    public String getSignName() {
+        return "PlayerDmgPlayer";
```

Comment:
Let's call it DamagePlayer to be consistent with DamageEntity, no need to specify "Player" at the start of player events since their purpose is being a player event